src/crypto/crypto_context.cc

```diff
@@ -47,10 +47,15 @@ static const char* const root_certs[] = {
 
 static const char system_cert_path[] = NODE_OPENSSL_SYSTEM_CERT_PATH;
 
-static X509_STORE* root_cert_store;
-
 static bool extra_root_certs_loaded = false;
 
+inline X509_STORE* GetOrCreateRootCertStore() {
+  static X509_STORE* store;
+  static uv_once_t once = UV_ONCE_INIT;
+  uv_once(&once, [] { store = NewRootCertStore(); });
```

Comment:
The uv_once_t isn’t really necessary, is it?

```suggestion
  static X509_STORE* store = NewRootCertStore();
```

Reply:
Mh – static initializers inside functions are initialized atomically on first function run: https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables

Reply:
Oh, like that! Yes, fair point; I'll make the change.

I thought we compile with `-fno-threadsafe-statics` but that's only libuv.